src/com/jwetherell/algorithms/strings/StringFunctions.java

```diff
@@ -21,6 +27,12 @@ public static final String reverseWithStringConcat(String string) {
     }
 
     public static final String reverseWithStringBuilder(String string) {
+
+        // Check if the input string is null or empty, and return an empty string if true
+        if (string == null || string.isEmpty()) {
```

Comment:
I added the the same if block here because this method also needs the same check as the previous regarding the null and empty string cases.